### DISCUSSION TITLE
Image alignment and code blocks

### DISCUSSION THREAD

**Discussion by julienguegan** [mmistakes/minimal-mistakes#3187] [Q&A]
### What happened?

- Minimal Mistakes version:
- Ruby gem or remote theme version:
- Jekyll version:
- Git repository URL:
- Hosted on GitHub Pages (if yes provide URL to site):
- Operating system:

### Expected behavior

Hello all,

I would like to position a code block and an image side by side. I have already succeed to place text and image next to each other with the tag `{: .align-right}` but associated with a code block it does not seems to work ... For example, if I write the following in a post : 

```

![image-right](/assets/images/gradient_descent.gif){: .align-right width="45%"}
'''python 
def gradient_descent(f, x0, alpha=0.01, itermax=1000):
    # initialization
    x, fx = np.zeros((itermax+1, len(x0))), np.zeros((itermax+1, 1))
    x[0,:] = x0
    # iterative loop
    k = 0
    while (k < itermax):
        grad_fxk = numerical_gradient(f, x[k,:], h=0.1)
        x[k+1,:] = x[k,:] - alpha * grad_fxk
        k = k+1
    return x
'''

```

The final rendering looks like this. The image does not appear and huge gap is created into the code block ...

![image](https://user-images.githubusercontent.com/45081450/132944329-731ed170-8502-4513-8f8a-76765ccd8b44.png)

**Does someone know if it is just impossible to do it or if there is a trick/possibility with HTML ?**

Thank you for your help !

### Steps to reproduce the behavior

Use `{: .align-right}` with a code block

**Comment by mmistakes** [comment]
It's definitely possible, but you're going to need to write some custom CSS and HTML to make it work.
You'll want to investigate how to use flexbox or the newer `display: grid` properties as it's not solution limited to Jekyll or this theme.

At a basic level you'll want structure your HTML like this:

```html
<div class="wrapper-name">
  <!-- code block html -->
  <!-- image html -->
</div>
```

Then you can do some CSS like:

```css
.wrapper-name {
  display: flex;
}
```

Which will squish the two next to each other. But you'll need more CSS to assign widths of each (I'd suggest using percentages), when they should wrap, gutter size, etc.

**Reply by julienguegan** [reply]
OK thank you for your answer. So I will need to have some CSS/HTML skills ... I will try a bit but if I struggle too much (as I don't have any experience in CSS) I will not continue it, I'll keep you informed

**Reply by mmistakes** [reply]
This is a good resource that explains all the flexbox properties. Honestly you won't need 99% of it.
I think if you wrap the 2 elements you want to align, give that a `display: flex` and then set the 2 elements inside of that to `flex: 1` that will get you close.

Then depending on how wide you want each you could do something like `width: 75%` on the code block element and  `width: 25%` for the image for a 3:1 ratio. Or go 50/50. Depend on how you want it all to look.

**Reply by mmistakes** [reply]
https://css-tricks.com/snippets/css/a-guide-to-flexbox/
